### PR TITLE
Randomized format set updates

### PR DESCRIPTION
**Pull request by ACakeWearingAHat**
Please merge before end of month if you can. No worries if you can't.

**Gen 9 Random Battle:**
-Chimecho now runs a third set of Cosmic Power, Stored Power, Dazzling Gleam, and Recover with Tera Steel.
-Bulky Attacker (non-Iron Defense) Probopass and Carbink no longer exist.
-Tera Blast Thundurus-Incarnate now gets Attack investment for when Defiant procs.
-Corviknight: +Pressure (rolled with Mirror Armor)
-Protect Swalot: +Sticky Hold (rolled with Liquid Ooze)
-Dragon Dance Lapras: -Freeze-Dry, +Icicle Spear (now gets Dice instead of Boots)
-QD Ribombee: -Bug Buzz, +Draining Kiss
-Swampert: +Yawn
-Empoleon: +Roar
-Sheer Force Cetitan: -Tera Fairy, -Play Rough, +Superpower. Will now always get Liquidation.
-Zamazenta-Crowned: +Tera Ghost
-Luvdisc: +Tera Ground
-Galarian Zapdos: +Tera Steel
-Setup Sweeper Flamigo: +Tera Steel
-Falinks: -Tera Ghost, -Tera Fighting
-Barraskewda: +Tera Water

Winrate-based reversions:
-Qwilfish is regaining Destiny Bond.
-Magearna is regaining Thunder Wave.
-Ampharos is losing Dragon Tail.
-Eiscue is regaining Zen Headbutt and Iron Head on non-Tera Blast and Liquidation on Tera Blast, thus also giving it Sitrus back.

Technical:
-Roar and Yawn can't generate together.
-A hardcode for Brute Bonnet was removed.
-A hardcode for Barraskewda was changed to a simpler, more straightforward one.

**Gen 9 Random Doubles**
-Lucario's sets have been changed to one mixed attacking set with Close Combat, Flash Cannon, Extreme Speed, and Protect and one Choice Band set with CC, Espeed, Mash, and Rock Slide.
-Smeargle's first set has been changed to Population Bomb, Spiky Shield, No Retreat, and Baton Pass with a Wide Lens. Smeargle's other set with Decorate now runs Sitrus Berry instead of Focus Sash.
-Hydrapple's sets have been changed significantly. It now has a Bulky Protect set with Pollen Puff, Fickle Beam, and Giga Drain/Leaf Storm, a Choice Specs Wallbreaker set with Earth Power, Draco Meteor, Leaf Storm, and Giga Drain, and an Assault Vest Bulky Attacker set with Fickle Beam, Giga Drain/Leaf Storm, Pollen Puff, and Earth Power. AV is only Tera Steel, the others are Tera Fire/Steel.
-Landorus: +Rock Slide (will not coincide with Nasty Plot)
-Falinks: +Rock Slide, +Protect, -Tera Fighting
-Bulky Protect Raikou: -Tera Grass, +Tera Water (Raikou will no longer get three electric moves)

**Gen 9 CAP Random Battle**
-This format will still exist after the spotlight has ended. This patch allows CAP rands to be created by adding the rule "+cap" to a Gen 9 Random Battle tournament or challenge. This has been tested and Hecate and Hizo helped me with the code.
-Level balancing has occurred. Caribolt, Kitsunoh, and Naviathan have lost two levels, and Stratagem, Crucibelle, Cresceidon, Volkraken, Miasmaw, and Voodoom have lost one level.
-Fast Attacker Kitsunoh: -Iron Fist, -Bullet Punch, -Tera Ghost, -Tera Steel, -Tera Fairy, +Trace, +U-turn (Now always gets Close Combat)
-Bulky Attacker Kitsunoh: -Iron Fist, +Trace, +Encore
-Voodoom: +Vacuum Wave
-Changes to Chuggalong will happen when its nerfs are implemented onsite.

**Gen 9 Baby Random Battle**
-Grass-type STAB is now enforced on all dual-typed Grass-types except Rowlet.
-Heavy-Duty Boots no longer exists in this format in any capacity.
-Nosepass now has a Bulky Attacker set with Volt Switch and Stealth Rock, in addition to its preexisting Iron Defense set.
-Drilbur has been split up between Rock Slide and Poison Jab; it now always gets Rapid Spin.
-Axew and Jangmo-o now run a second set each with Eviolite Swords Dance + Scale Shot.
-Tera Blast Drilbur no longer exists. Drilbur's other set no longer runs Throat Chop or Tera Fairy and will now always get Stone Edge.
-Acrobatics Pikipek no longer exists.
-Tinkatink's role has been change to Bulky Attacker. It will now always have Knock Off.
-Sewaddle: -Electroweb, +Iron Defense (Iron Defense will only be rolled if the team already has a sticky web user.)
-Slowpoke: All moves removed except Curse, Slack Off, Liquidation, Zen Headbutt, and Thunder Wave.
-BD Magby: -Temper Flare, +Fire Punch
-Fennekin: +Tera Ground, +Mud Shot
-Noibat: -Air Slash
-Minccino: -Triple Axel, -Tera Ice
-Feebas: -Tera Normal, -Double Edge, +Tera Dragon, +Scale Shot
-Surskit, Deerling, and Spinarak: +Tera Ghost
-Setup Psyduck: +Tera Grass
-Horsea and Tera Blast Poliwag: +Tera Dragon
-Tera Blast Sinistea: +Tera Fairy
-Sandile: +Tera Flying
-Tera Blast Zorua-H: +Tera Fighting
-Bulky Attacker Zorua: -Tera Dark
-Crabrawler: +Tera Electric
-Varoom: -Tera Steel, -Tera Poison
-Tinkatink: -Tera Fairy
-Charcadet: -Tera Fire
-Wattrel: -Tera Electric
-Seel, Finneon: -Tera Water
-Phanpy, Mudbray: -Tera Ground
-Setup Sweeper Gimmighoul-Roaming: -Tera Fairy, +Tera Rock
-Exeggcute: -Tera Dark, +Tera Steel
-Grimer: +Tera Dark
-Fast Attacker Scorbunny: -Tera Poison, -Tera Fighting
-Sprigatito: -Tera Fairy

**Old Gens:**
-In Gens 4-7, Magmortar no longer runs Hidden Power Grass or Substitute and can now run Taunt.
-In Gens 5-7, Maractus no longer runs Sucker Punch.
-In Gens 5-7, Lucario no longer runs Dark Pulse.
-In Gens 6-7, Mega Ampharos now has a set split where one set runs Agility and Thunderbolt and the other runs Discharge and RestTalk or Heal Bell.
-In Gens 6-7, Mega Absol has lost Fire Blast, gained Iron Tail, and merged its sets together
-In Gens 6-7, Espeon now always gets Dazzlilng Gleam.
-In Gen 7, Meloetta now runs a Z-Celebrate set with its usual three special attacks.
-In Gen 7, Solgaleo now always runs Earthquake.
-In Gen 7, Lycanroc and Lycanroc-Dusk now run Zen Headbutt instead of Fire Fang.
-In Gen 6, AV Hariyama can sometimes roll Guts on its AV Pivot set.
-In Gen 4, Tropius no longer runs Roost on its setup set.
-In Gen 4, CM Lunatone now runs Signal Beam instead of Shadow Ball.
-In Gen 4, Plusle and Minun now always run HP Ice.
-In Gen 4, Hypno now has a third set with Nasty Plot and Baton Pass.
-In Gen 3, Spikes is now enforced on all sets that have the move.
-In Gen 3, Deoxys (normal) no longer runs Spikes.
-In Gen 3, Forretress's sets were merged because Spikes is forced now and we wouldn't have to worry about Choice Band existing anymore.
-In Gen 3, Blissey no longer runs its setup set.
-In Gen 3, Magmar no longer runs Psychic and non-setup Magmar can now run Toxic.
-In Gen 3, Wallbreaker Rayquaza now always runs Extreme Speed.
-In Gen 2, Forretress's double-edge set no longer exists.
-In Gen 2, Porygon2's non-Curse set no longer exists.
-In Gen 2, Aipom now runs a second set with RestTalk, Curse, and Return.
-In Gen 2, Crobat now runs Wing Attack instead of Return and always runs HP Ground.
-In Gen 2, Magcargo's Generalist role has been changed to Bulky Attacker to enforce a second attack and that set now has Earthquake. Its other set now runs Fire Blast or Flamethrower instead of Sleep Talk.
-In Gen 2, Fast Attacker Butterfree now runs Reflect instead of Substitute
-In Gen 2, Belly Drum Quagsire no longer exists and Bulky Attacker Quagsire now sometimes runs Sludge Bomb.

### CODE REVIEW

**Comment by ACakeWearingAHat** [comment]
please merge and patch alongside #10520

**Comment by ACakeWearingAHat** [comment]
This PR is now tied to both #10522 and #10520. Please merge all three at the same time.

File: sim/teams.ts [hunk 1]
```diff
@@ -624,7 +624,7 @@ export const Teams = new class Teams {
 			TeamGenerator = require(`../data/mods/gen9ssb/random-teams`).default;
 		} else if (toID(format).includes('gen9babyrandombattle')) {
 			TeamGenerator = require(`../data/random-battles/gen9baby/teams`).default;
-		} else if (toID(format).includes('gen9caprandombattle')) {
+		} else if (toID(format).includes('gen9randombattle') && format.customRules?.includes('+cap')) {
```
Comment (KrisXV) [inline sim/teams.ts:627]:
```suggestion
		} else if (toID(format).includes('gen9randombattle') && format.ruleTable?.has('+pokemontag:cap')) {
```

File: data/random-battles/gen9/teams.ts [hunk 2]
```diff
@@ -316,6 +316,10 @@ export class RandomTeams {
 	 * Doesn't count bans nested inside other formats/rules.
 	 */
 	private hasDirectCustomBanlistChanges() {
+		if (
+			this.format.customRules &&
+			this.format.customRules.length === 1 && ['+cap'].includes(this.format.customRules[0])
+		) return false;
```
Comment (KrisXV) [inline data/random-battles/gen9/teams.ts:322]:
```suggestion
		if (this.format.ruleTable?.has('+pokemontag:cap')) return false;
```

File: data/random-battles/gen9cap/sets.json [hunk 3]
```diff
@@ -102,18 +102,18 @@
...
-                "teraTypes": ["Fairy", "Fighting", "Ghost", "Steel"]
+                "movepool": ["Close Combat", "Defog", "Encore", "Meteor Mash", "Poltergeist", "Strength Sap", "Trick", "U-turn", "Will-O-Wisp"],
+                "abilities": ["Trace"],
+                "teraTypes": ["Fighting"]
             },
             {
                 "role": "Bulky Attacker",
-                "movepool": ["Defog", "Meteor Mash", "Poltergeist", "Strength Sap", "U-turn", "Will-O-Wisp"],
-                "abilities": ["Iron Fist"],
+                "movepool": ["Defog", "Encore", "Meteor Mash", "Poltergeist", "Strength Sap", "U-turn", "Will-O-Wisp"],
```
Comment (KrisXV) [inline data/random-battles/gen9cap/sets.json:115]:
```suggestion
                "movepool": ["Defog", "Meteor Mash", "Poltergeist", "Strength Sap", "U-turn", "Will-O-Wisp"],
```

**Pull request merged**